src/network_manager.rs

```diff
@@ -5,17 +5,21 @@
 //! simple ownership semantics of reading and writing to the network stack. This allows the network
 //! stack to be used to transmit buffers that may be stored internally in other structs without
 //! violating Rust's borrow rules.
+use core::mem;
+
 use embedded_nal::{nb, SocketAddr, TcpClientStack};
+use heapless::Vec;
 
 use crate::Error;
 
 /// Simple structure for maintaining state of the network connection.
-pub(crate) struct InterfaceHolder<TcpStack: TcpClientStack> {
+pub(crate) struct InterfaceHolder<TcpStack: TcpClientStack, const MSG_SIZE: usize> {
     socket: Option<TcpStack::TcpSocket>,
     network_stack: TcpStack,
+    unfinished_packet: Option<Vec<u8, MSG_SIZE>>,
```

Comment:
Instead of buffering the message here, why don't we leverage the `pending_publish()` vector in `SessionState`? We already have data there, and oculd potentially replace the LinearMap with something like a `heapless::Pool` of messages. Then, we can just allocate a buffer from the pool for storing a pending unfinished packet.

This has the benefit of letting the user tailor the buffering needs to the RAM avialability of their target.

Reply:
After looking into this more, I'm more convinced this approach that you have is correct. The session state buffers are for QoS messages, and mixing in a partially written message with those messages doesn't make sense to me.